src/plugins/saved_objects_management/server/routes/scroll_count.ts

```diff
@@ -56,18 +55,15 @@ export const registerScrollForCountRoute = (router: IRouter) => {
         findOptions.hasReferenceOperator = 'OR';
       }
 
-      const objects = await findAll(client, findOptions);
-
-      const counts = objects.reduce((accum, result) => {
-        const type = result.type;
-        accum[type] = accum[type] || 0;
-        accum[type]++;
-        return accum;
-      }, {} as Record<string, number>);
-
+      const counts: Record<string, number> = {};
```

Comment:
NIT: We may be able to convert this to an ES aggregation, so there's no need to loop through all documents.

WDYT?

Reply:
Yeah, I came to the same conclusion *after* I adapted the code. I wanted to try it, but in a follow-up (this one was initially supposed to be only for the usage collection calls)